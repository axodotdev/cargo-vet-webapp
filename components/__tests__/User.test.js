import { mount } from "@vue/test-utils";
import AxoButton from "@axodotdev/fringe/lib/Button.vue";
import User from "../User.vue";

let wrapper;

const createComponent = ({ props = {}, shallow = false } = {}) => {
  wrapper = mount(User, {
    props,
    shallow,
  });
};

afterEach(() => {
  wrapper.unmount();
  wrapper = null;
});

describe("User component", () => {
  const user = {
    login: "sara",
    avatar_url: "https://test.com",
  };

  beforeEach(() => {
    createComponent({ props: { user } });
  });

  // I think we can remove this given that the more detailed
  // checks also checks it mounts and snapshots can be difficult,
  // but I wanted to show what the transformation looks like before
  // just deleting it
  it("renders component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders the user's name", () => {
    expect(wrapper.find("p").text()).toBe(`Welcome ${user.login}`);
  });

  it("renders a logout button", () => {
    expect(wrapper.findComponent(AxoButton).exists()).toBe(true);
    expect(wrapper.findComponent(AxoButton).text()).toBe("Logout");
  });
});
