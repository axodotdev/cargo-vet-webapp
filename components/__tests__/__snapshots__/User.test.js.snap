// Vitest Snapshot v1

exports[`User component > renders component 1`] = `
"<div class=\\"items-center gap-2 p-5 flex flex-col\\">
  <div class=\\"flex gap-4 items-center mb-4\\"><img src=\\"https://test.com\\" class=\\"rounded-full w-10 h-10\\">
    <p class=\\"m-0\\"> Welcome <span class=\\"font-bold\\">sara</span></p>
  </div><button class=\\"bg-axo-orange text-slate-900 w-full rounded-3xl pointer m-1 text-2xl py-1 hover:bg-axo-pink transition\\"> Logout </button>
</div>"
`;
